### PR TITLE
FEATURE: CL-11346 Added validElements resource

### PR DESCRIPTION
**Pull request by vojty**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/359/ : SUCCESS in 2m 00s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/124/ : SUCCESS in 1m 14s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/360/ : SUCCESS in 2m 26s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/125/ : SUCCESS in 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/361/ : SUCCESS in 1m 00s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/126/ : SUCCESS in 1m 25s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/362/ : SUCCESS in 46s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/127/ : SUCCESS in 30s

File: src/metadata.js [hunk 1]
```diff
@@ -543,3 +544,38 @@ export function getObjectUri(projectId, identifier) {
...
+ *      - restrictiveDefinition {String}
+ * @return {Object} ValidElements response with:
+ *      - items {Array} elements
+ *      - paging {Object}
+ *      - elementsMeta {Object}
+ */
+export function getValidElements(projectId, identifier, options = {}) {
+    const query = pick(options, ['limit', 'offset', 'order', 'filtering', 'prompt']);
+    const queryParams = Object.keys(query)
+        .map(option => `${option}=${query[option]}`)
```
Comment (bredy) [inline src/metadata.js]: Pls add encodeURIComponent to query[option]

File: src/metadata.js [hunk 2]
```diff
@@ -543,3 +544,38 @@ export function getObjectUri(projectId, identifier) {
...
+ *      - complement {Boolean}
+ *      - includeTotalCountWithoutFilters {Boolean}
+ *      - restrictiveDefinition {String}
+ * @return {Object} ValidElements response with:
+ *      - items {Array} elements
+ *      - paging {Object}
+ *      - elementsMeta {Object}
+ */
+export function getValidElements(projectId, identifier, options = {}) {
+    const query = pick(options, ['limit', 'offset', 'order', 'filtering', 'prompt']);
```
Comment (bredy) [inline src/metadata.js]: "filter" instead of "filtering"

File: src/metadata.js [hunk 3]
```diff
@@ -543,3 +544,38 @@ export function getObjectUri(projectId, identifier) {
...
+ *      - uris {Array}
+ *      - complement {Boolean}
+ *      - includeTotalCountWithoutFilters {Boolean}
+ *      - restrictiveDefinition {String}
+ * @return {Object} ValidElements response with:
+ *      - items {Array} elements
+ *      - paging {Object}
+ *      - elementsMeta {Object}
+ */
+export function getValidElements(projectId, identifier, options = {}) {
```
Comment (bredy) [inline src/metadata.js]: Please rename "identifier" to "id" -  "identifier" could be confusing.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/364/ : SUCCESS in 57s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/129/ : SUCCESS in 1m 50s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/9/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/365/ : SUCCESS in 2m 02s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/130/ : SUCCESS in 41s

**Pull request merged**